### DISCUSSION TITLE
How to list all valid certificates?

### DISCUSSION THREAD

**Discussion by Artur-nav** [strongswan/strongswan#705] [Q&A]
Let's say we issued total 50 X.509 certificates, 20 of them expired, another 10 has been revoked - how to list 20 remaining certificates, that are still valid? Command "ipsec listcerts" lists 11 certificates and it is correct according to a documentation: listcerts shows only loaded certificates, so it won't show 9 ones that has not been used from last VPN server start.

How to display a complete list of these 20 valid certificates, not just 11 that belong to active users?

**Comment by tobiasbrunner** [comment]
How should the IKE daemon magically list certificates it has never seen and doesn't know anything about? Seems like you have to rethink your PKI management.

**Reply by Artur-nav** [reply]
I'm using "ipsec pki" commands to generate user key and certificate:
private/"$FIRSTNAME$LASTNAME"_Key.der 
certs/"$FIRSTNAME$LASTNAME"_Cert.der

That's why I thought "ipsec listcerts" would find certificates generated by itself.

**Reply by tobiasbrunner** [reply]
`ipsec pki` is just an alias for `pki` (mostly for legacy reasons or if the latter is not in your `PATH`), it has absolutely nothing to do with the `ipsec` script or the rest of its "commands", which are basically aliases to other tools, see [ipseccommand](https://wiki.strongswan.org/projects/strongswan/wiki/ipseccommand).

If you want to check if a certificate is valid with `pki` you could use [`pki --verify`](https://wiki.strongswan.org/projects/strongswan/wiki/IpsecPkiVerify).
